### ISSUE TITLE
New warning in 'languageserver' (from linter exclusion?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1987]
Haven't confirmed this is a regression yet, but we observe this warning on main that is not present on v3.0.2:

```
Could not find linter named ‘%s’ in the list of active linters. Make sure the linter is uniquely identified by the given name or prefix.
```

**Comment by MichaelChirico** [comment]
First guess is it's this:

https://github.com/REditorSupport/languageserver/blob/1e71561a41ef59126273f16bbb466c0b30a2aa64/R/code_action.R#L102

So we're getting dinged for doing text-based matching of the `# nolint` directive.

**Comment by MichaelChirico** [comment]
The issue is `lintr::lint()` comes up clean for {languageserver} under v3.0.2, so the code throwing the warning is never executed.

But `indentation_linter()` has a lot of hits at HEAD of {languageserver}, hence it's running the exclusion code trying to figure out if the lints should be skipped.

To confirm this, I edited R/code_action.R to use `=` and throw `assignment_linter()`, and indeed the same warning is thrown.

Since this is not a new warning, I'm going to close the issue, re-filing as a more specific issue (but not including it in the 3.1.0 milestone).
